src/controllers/Auth.ts

```diff
@@ -0,0 +1,529 @@
+        this.kuzzle,
+        response.result._id,
+        response.result._source));
+  }
+
+  /**
+   * Get credential information of the specified strategy for the current user.
+   *
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/auth/get-my-credentials
+   *
+   * @param strategy Name of the strategy to use
+   * @param options Additional options
+   *    - "queuable" If true, queues the request during downtime, until connected to Kuzzle again
+   *
+   * @returns An object representing the credentials for the provided authentication strategy.
+   *    Its content depends on the authentication strategy.
+   */
+  getMyCredentials(
+    strategy: string,
+    options: { queuable?: boolean } = {}
+  ): Promise<JSONObject> {
+    return this.query({
+      strategy,
+      action: 'getMyCredentials'
+    }, options)
+      .then(response => response.result);
+  }
+
+  /**
+   * Gets the rights array of the currently logged user.
+   *
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/auth/get-my-rights
+   *
+   * @param options Additional options
+   *    - "queuable" If true, queues the request during downtime, until connected to Kuzzle again
+   *
+   * @returns An array containing user rights objects
+   */
+  getMyRights (
+    options: { queuable?: boolean } = {}
+  ): Promise<Array<{
+    /**
+     * Controller on wich the rights are applied
+     */
+    controller: string,
+    /**
+     * Action on wich the rights are applied
+     */
+    action: string,
+    /**
+     * Index on wich the rights are applied
+     */
+    index: string,
+    /**
+     * Collection on wich the rights are applied
+     */
+    collection: string,
+    /**
+     * Rights ("allowed" or "denied")
+     */
+    value: string
+  }>> {
+    return this.query({
+      action: 'getMyRights'
+    }, options)
+      .then(response => response.result.hits);
+  }
+
+  /**
+   * Get all the strategies registered in Kuzzle by all auth plugins
+   *
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/auth/get-strategies
+   *
+   * @param options Additional options
+   *    - "queuable" If true, queues the request during downtime, until connected to Kuzzle again
+   *
+   * @returns An array of available strategies names
+   */
+  getStrategies (options: { queuable?: boolean } = {}): Promise<Array<string>> {
+    return this.query({
+      action: 'getStrategies'
+    }, options)
+      .then(response => response.result);
+  }
+
+  /**
+   * Send login request to kuzzle with credentials
+   * If login success, store the jwt into kuzzle object
+   *
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/auth/login
+   *
+   * @param strategy Name of the strategy to use
+   * @param credentials Credentials object for the strategy
+   * @param expiresIn Expiration time in ms library format. (e.g. "2h")
+   *
+   * @returns The encrypted JSON Web Token
+   */
+  login (
+    strategy: string,
+    credentials: JSONObject,
+    expiresIn: string = null
```

Comment:
[expiresIn](https://docs.kuzzle.io/core/2/api/controllers/auth/login/) can be a number. Also, it's optional:

```suggestion
    expiresIn?: string|number = null
```

(I _think_ that this works, but I'm fairly new to TS)